uniswap_swap_detection/src/agent.spec.ts

```diff
@@ -0,0 +1,244 @@
+import {
+  FindingType,
+  FindingSeverity,
+  Finding,
+  HandleTransaction,
+  ethers,
+} from "forta-agent";
+import { utils, Contract } from "ethers";
+
+import agent, { provideHandleTransaction } from "./agent";
+import { SWAP_EVENT } from "./constant";
+import { createAddress } from "forta-agent-tools";
+import {
+  MockEthersProvider,
+  TestTransactionEvent,
+} from "forta-agent-tools/lib/test";
+
+const iface: utils.Interface = new utils.Interface([
+  "function token1() external view returns (address)",
+  "function token0() external view returns (address)",
+  "function fee() external view returns (uint24)",
+]);
+
+export const CREATED_POOL =
+  "event PoolCreated(address indexed token0,address indexed token1,uint24 indexed fee,int24 tickSpacing,address pool)";
+describe("swap occur", () => {
+  let handleTransaction: HandleTransaction;
+
+  let txEvent: TestTransactionEvent;
+
+  const address1 = "0x61D3f523cd7e93d8deF89bb5d5c4eC178f7CfE76".toLowerCase();
+  const token_1 = "0xc2132D05D31c914a87C6611C10748AEb04B58e8F";
+
+  const token_0 = "0x765Af38A6e8FDcB1EFEF8a3dd2213EFD3090B00F";
+
+  const fee_ = 3000;
+  const mockProvider: MockEthersProvider = new MockEthersProvider();
+  mockProvider.setNetwork(137);
+
+  handleTransaction = provideHandleTransaction(mockProvider as any);
+
+  // const PROVIDER = getEthersProvider()
+  beforeAll(() => {
+    jest.spyOn(Date, "now").mockImplementation(() => 1590000000000);
+  });
```

Comment:
Not exactly what this is for. Do you mind explaining?

Reply:
i need to mock a new date when i receive an error in regards to date when i run my test